check_sites.py

```diff
@@ -18,33 +18,63 @@
 import os
 from publicsuffix2 import PublicSuffixList
 
+def find_diff_sets(old_sets, new_sets):
+    """Finds changes made between two dictionaries of First-Party Sets
+
+        Finds First-Party Sets that have been added or modified in old_sets
+        to create new_sets and returns them as the dictionary diff_sets. 
+        Additionally, finds First-Party Sets that have been removed from
+        old_sets to create new_sets and returns them as subtracted_sets.
+
+        Args:
+            old_sets: a dictionary of primary->FpsSet
+            new_sets: a dictionary of primary->FpsSet
+        Returns:
+            diff_sets: a dictionary of primary->FpsSet
+            substracted_sets: a dictionary of primary->FpsSet
+    """
+    diff_sets = {}
+    for primary in new_sets:
+        if primary not in old_sets:
+            diff_sets[primary] = new_sets[primary]
+        elif new_sets[primary] != old_sets[primary]:
+            diff_sets[primary] = new_sets[primary]
```

Comment:
We can simplify the loop and the internal if statement here, with [`dict.items`](https://docs.python.org/3/library/stdtypes.html#dict.items):

```py
diff_sets = {}
for primary, fps in new_sets.items():
    if primary not in old_sets || fps != old_sets[primary]:
        diff_sets[primary] = fps
```

We can then simplify the if statement further, taking advantage of [`dict.get`](https://docs.python.org/3/library/stdtypes.html#dict.get):

```py
diff_sets = {}
for primary, fps in new_sets.items():
    if fps != old_sets.get(primary):
        diff_sets[primary] = fps
```

I would then probably write this using a [dictionary comprehension](https://www.programiz.com/python-programming/dictionary-comprehension):

```py
diff_sets = {primary: fps
             for primary, fps in new_sets.items()
             if fps != old_sets.get(primary)
            }
```

Declarative descriptions of data are much easier to understand than imperative definitions that build the data over time, IMO.

Reply:
Done